src/sparseml/pytorch/recipe_template/main.py

```diff
@@ -0,0 +1,351 @@
+
+        del pruning_args["init_sparsity"]
+        del pruning_args["final_sparsity"]
+        del pruning_args["inter_func"]
+
+    elif pruning_algo == "mfac":
+        modifier_class = MFACPruningModifier
+        extra_recipe_variables.update(dict(num_grads=64, fisher_block_size=10000))
+        pruning_args.update(
+            dict(
+                num_grads="eval(num_grads)", fisher_block_size="eval(fisher_block_size)"
+            )
+        )
+
+    elif pruning_algo == "movement":
+        modifier_class = MovementPruningModifier
+        # movement pruning does not support global_sparsity
+        del pruning_args["global_sparsity"]
+
+    elif pruning_algo == "obs":
+        modifier_class = OBSPruningModifier
+        extra_recipe_variables.update(dict(num_grads=1024, fisher_block_size=50))
+        pruning_args.update(
+            dict(
+                num_grads="eval(num_grads)",
+                fisher_block_size="eval(fisher_block_size)",
+            )
+        )
+    elif pruning_algo == "constant":
+        modifier_class = ConstantPruningModifier
+        pruning_args = dict(start_epoch=0.0, end_epoch=10.0, params="__ALL_PRUNABLE__")
+    else:
+        raise ValueError(f"Unknown pruning_algo type {pruning_algo}")
+
+    pruning_modifier = ModifierYAMLBuilder(
+        modifier_class=modifier_class, **pruning_args
+    )
+    return [pruning_modifier], extra_recipe_variables
+
+
+def _get_prunable_params_for_module(model: Module) -> List[str]:
+    return [
+        f"{name}.weight"
+        for name, _ in get_prunable_layers(module=model)
+        if "input" not in name and "output" not in name
+    ]
+
+
+def _get_quantizable_params_for_module(model: Module) -> List[str]:
+    return [name for name, _ in get_quantizable_layers(module=model)]
+
+
+def _get_quantization_modifier_builders(
+    post_pruning: bool,
+    target: str,
+    model: Optional[Module] = None,
+) -> Tuple[List[ModifierYAMLBuilder], Dict[str, Any]]:
+    extra_recipe_variables = {}
+    freeze_graph_epoch = (
+        "eval(_num_pruning_epochs + num_qat_epochs - num_qat_finetuning_epochs)"
+        if post_pruning
+        else "eval(num_qat_epochs - num_qat_finetuning_epochs)"
+    )
+    quantization_modifier = ModifierYAMLBuilder(
+        modifier_class=QuantizationModifier,
+        start_epoch="eval(_num_pruning_epochs)" if post_pruning else 0.0,
+        submodules="eval(quantization_submodules)",
+        disable_quantization_observer_epoch=freeze_graph_epoch,
+        freeze_bn_stats_epoch=freeze_graph_epoch,
+        tensorrt=target == "tensorrt",
+        quantize_linear_activations=False,
+        quantize_conv_activations=False,
+        quantize_embedding_activations=False,
+        exclude_module_types=["LayerNorm", "Tanh"],
+    )
+
+    extra_recipe_variables["quantization_submodules"] = (
+        "null" if not model else _get_quantizable_params_for_module(model=model)
+    )
```

Comment:
we don't want to do this as doing haphazardly can lead to things like conv-bn-relu blocks being broken up and not properly fused. - just leave none to quantize the entire model by default for now. if a user wants to break out what parts of the graph are quantized, they can update based on their understanding of their model flow